src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1817,7 +1808,22 @@ public bool IsOnTypeHeader(SyntaxNode root, int position)
 
         public bool IsInPropertyDeclarationHeader(SyntaxNode node)
         {
-            var propertyDeclaration = node.GetAncestor<PropertyDeclarationSyntax>();
+            var propertyDeclaration = node.GetAncestorOrThis<PropertyDeclarationSyntax>();
+            if (propertyDeclaration == null)
+            {
+                return false;
+            }
+
+            var start = propertyDeclaration.AttributeLists.LastOrDefault()?.GetLastToken().GetNextToken().SpanStart ??
+                        propertyDeclaration.SpanStart;
```

Comment:
this logic feels duplicative.  can we extract this into a helper "GetStartOfNodeExcludingAttributes"?

Reply:
I would love to but IIRC there's no Interface saying `IHasAttributesSyntax` or smth *) and so I have no way to generic over the shape "has `.AttributeLists`".

I could refactor smth like "GetLocationAfterAttributesLists(this AttributeLists, ...)"

*) I'll look once more, mightve missed it

Reply:
> I would love to but IIRC there's no Interface saying

Sure tehre is:  SyntaxGenerator.GetAttributes :)